src/directives/model/default.js

```diff
@@ -31,7 +34,7 @@ module.exports = {
     // shared setter
     function set () {
       self.set(
-        number ? _.toNumber(el.value) : el.value,
+        file ? el.files : ( number ? _.toNumber(el.value) : el.value ),
```

Comment:
The point is Vue will try to convert the object properties into getter/setters and they may or may not break these native objects. And if we skip conversion for these objects, they won't be able to be used for data-binding.